views/post.py

```diff
@@ -68,3 +68,65 @@ def create_post(post):
         conn.commit()
 
         return db_cursor.lastrowid
+
+
+def get_post(pk, query_params=None):
+    if query_params and "_expand" in query_params:
+        with sqlite3.connect("./db.sqlite3") as conn:
+            conn.row_factory = sqlite3.Row
+            db_cursor = conn.cursor()
+
+            db_cursor.execute(
+                """
+                SELECT
+                    p.id AS post_id,
+                    p.title AS post_title,
+                    p.content AS post_content,
+                    p.publication_date AS publication_date,
+                    u.id AS user_id,
+                    u.username AS username
+                FROM Posts p
+                LEFT JOIN Users u ON u.id = user_id
+                WHERE p.id = ?
+                """,
+                (pk,),
+            )
+            query_results = db_cursor.fetchone()
+
+            if query_results:
+                serialized_post = json.dumps(dict(query_results))
+                return serialized_post
+            else:
+                return json.dumps({"error": "Post not found"})
+
+    else:
+        with sqlite3.connect("./db.sqlite3") as conn:
```

Comment:
The issue ticket also mentions that the header url should be provided, so just add p.image_url into your list of column names in the SELECT statement and I think that should fix the issue. 
Other than that, this section all looks good to me and worked when tested!